packages/core/core/src/BundleGraph.js

```diff
@@ -812,16 +814,25 @@ export default class BundleGraph {
   }
 
   getBundlesInBundleGroup(bundleGroup: BundleGroup): Array<Bundle> {
-    return this._graph
-      .getNodesConnectedFrom(
-        nullthrows(this._graph.getNode(getBundleGroupId(bundleGroup))),
-        'bundle',
-      )
-      .filter(node => node.type === 'bundle')
-      .map(node => {
-        invariant(node.type === 'bundle');
-        return node.value;
-      });
+    return (
+      this._graph
+        .getNodesConnectedFrom(
+          nullthrows(this._graph.getNode(getBundleGroupId(bundleGroup))),
+          'bundle',
+        )
+        .filter(node => node.type === 'bundle')
+        .map(node => {
+          invariant(node.type === 'bundle');
+          return node.value;
+        })
+        // Sort by bundleIds but reversed because the preorder DFS traversal in
+        // the bundler means that dependencies are added last (and need to run first)
+        .sort(
+          (a, b) =>
+            bundleGroup.bundleIds.indexOf(b.id) -
+            bundleGroup.bundleIds.indexOf(a.id),
+        )
+    );
```

Comment:
Would it be faster to do something like this?

```js
bundleGroup.bundleIds.map(id => nullthrows(this._graph.getNode(id)).value)
```

Reply:
Yeah, my worry was that `bundleGroup.bundleIds` and `getNodesConnectedFrom` could become out of sync (which actually was the case)

Reply:
Done in 3265bb7315075a3eb3fc61dc70804b5a82a21fa2, I've had to do some other changes to the bundle graph to keep `bundleIds` in sync with the graph